joern-cli/frontends/javasrc2cpg/src/test/scala/io/joern/javasrc2cpg/querying/GenericsTests.scala

```diff
@@ -4,195 +4,296 @@ import io.joern.javasrc2cpg.testfixtures.JavaSrcCode2CpgFixture
-      |    public static void testWildCardLower(Box<? super Integer> b) {
-      |        System.out.println(b);
-      |    }
-      |}
-      |
-      |
-      |// inheritsFrom Box
-      |public class Test extends Box<String> {}
-      |""".stripMargin)
+    "use erased types for the method full name in the constructor invocation" in {
+      cpg.call.nameExact("<init>").methodFullName.l shouldBe List("box.Box.<init>:<unresolvedSignature>(1)")
+    }
 
-  "it should create the correct generic typeDecl name" in {
-    cpg.typeDecl.nameExact("Box").l match {
-      case decl :: Nil => decl.fullName shouldBe "Box"
+    "use erased types for the method full name in the get method invocation" in {
+      cpg.call.nameExact("get").methodFullName.l shouldBe List("box.Box.get:<unresolvedSignature>(0)")
+    }
 
-      case res => fail(s"Expected typeDecl Box but got $res")
+    "not include generic types in type full names of objects" in {
+      cpg.method.name("test").local.name("b").typeFullName.l shouldBe List("box.Box")
     }
   }
 
-  "it should default to Object for a simple generic type" in {
-    cpg.method.name("getItem").l match {
-      case method :: Nil =>
-        method.fullName shouldBe "Box.getItem:java.lang.Object()"
-        method.signature shouldBe "java.lang.Object()"
+  "generic methods" should {
+    val cpg = code("""package foo;
+                     |
+                     |class Foo {
+                     |  public <S, T> T foo(S s) { return null; }
+                     |
+                     |  static void test(Foo f) {
+                     |      f.<Integer, String>foo(0);
+                     |  }
+                     |}
+                     |""".stripMargin)
+
+    "use erased types in the method fullName" in {
+      cpg.method.name("foo").fullName.l shouldBe List("foo.Foo.foo:java.lang.Object(java.lang.Object)")
+    }
 
-      case res => fail(s"Expected method getItem but got $res")
+    "use erased types in the call methodFullName" in {
+      cpg.method.name("test").call.name("foo").methodFullName.l shouldBe List(
+        "foo.Foo.foo:java.lang.Object(java.lang.Object)"
+      )
     }
   }
 
-  "it should default to Object for simple generic parameters" in {
-    cpg.method.name("setItem").l match {
-      case method :: Nil =>
-        method.fullName shouldBe "Box.setItem:void(java.lang.Object)"
-        method.signature shouldBe "void(java.lang.Object)"
+  "methods returning parameterized types" should {
+    val cpg = code("""package foo;
+                     |
+                     |class Box<T> {
+                     |  public <S> Box<S> into() { return null; }
+                     |
+                     |  public T get() { return null; }
+                     |
+                     |  static void test(Box<String> stringBox) {
+                     |    stringBox.<Integer>into().get();
+                     |  }
+                     |}
+                     |""".stripMargin)
```

Comment:
Is this just a smoke test? If so please add a comment stating this.

Reply:
Ah, I think this is a test I was halfway through writing before moving over to something else. I'll add the rest of the test now. It's not strictly speaking relevant to this PR, but is a gap in our generic testing